src/Util/ColumnParser.php

```diff
@@ -50,8 +50,7 @@ public function parseFields($arguments)
                     $type = 'primary';
                 }
             }
-
-            $nullable = (bool)preg_match('/\w+\?(\[\d+\])?/', $type);
+            $nullable = (bool)preg_match('/(\w+\?)(?:(\[(([1-9][0-9]|[1-9]){1,2})(?:(\,([1-9][0-9]|[1-9]))*)\]))*/', $type);
```

Comment:
Why does the nullable check have to be anything else than testing for a `?` after the field type in the first place (like `/^\w+\?/)`? I mean, whatever comes after that shouldn't be a concern as far as the nullability goes, or should it?

Reply:
In all fairness, `strpos` check is much faster that `preg_match`. Preg_match was there since the time I forked so kept using preg_match.
The code was reviewed above and I had a feedback to match the possible combinations.
Made the change to get it more stricter.
well yea, even I don't get why we match anything before `(\w+\?)`...
What do you advice?

Reply:
> Preg_match was there since the time I forked so kept using preg_match

That's why I'm asking, I was wondering why it was there in the first place ;)

Looking at the history and the column parsing regex, unless I'm missing something, there seems to be no need for a regex here at all, a simple `strpos()` check should suffice, given that neither `$regexpParseColumn`, nor `$regexpParseField` (in combination with  `$regexpParseColumn`) do really allow for `?` usage anywhere other than once after the field type, and the nullable check was also never actually used for any kind of format validation to begin with, a non-matching format would've just caused the nullability to not be determined.

So I'd say go with a relaxed `strpos()` check. If we want stricter matching, then we should start with checking whether applying `$regexpParseColumn` actually succeeds.

Reply:
Yea, will apply a relaxed strpos because checking `!==0` is not relevant to the use case here. Also, will keep the type cast as `(bool)` ... you know... just incase. ;)